client/pkg/jsonmessage/jsonmessage.go

```diff
@@ -189,16 +192,17 @@ func (jm *JSONMessage) Display(out io.Writer, isTerminal bool) error {
 	return nil
 }
 
-type JSONMessagesStream iter.Seq2[JSONMessage, error]
+// JSONMessagesStream is a stream of typed JSON messages.
+type JSONMessagesStream[A any] iter.Seq2[JSONMessage[A], error]
 
 // DisplayJSONMessagesStream reads a JSON message stream from in, and writes
-// each [JSONMessage] to out.
+// each [JSONMessage[A]] to out.
 // see DisplayJSONMessages for details
-func DisplayJSONMessagesStream(in io.Reader, out io.Writer, terminalFd uintptr, isTerminal bool, auxCallback func(JSONMessage)) error {
+func DisplayJSONMessagesStream[A any](in io.Reader, out io.Writer, terminalFd uintptr, isTerminal bool, auxCallback func(JSONMessage[A])) error {
```

Comment:
We could possibly internalise this if we want to keep the old signature (roughly); still need to look if we still need these, or if they need to be rewritten.